### PR TITLE
More tokens

### LINKED ISSUE #16: Add all token types
- [x] Operations

### PR DESCRIPTION
**Pull request by JakeRoggenbuck**
fixes #16

### CODE REVIEW

File: src/lexer/token.h [hunk 1]
```diff
@@ -17,6 +17,73 @@ typedef enum {
     TT_CBRACKET, // ]
     TT_SEMI, // ;
     TT_NO_TOKEN, // Not a token
+
```
Comment (adamhutchings) [inline src/lexer/token.h]: I thought of a couple more that need to be added:
```c
TT_XOR, // ^
TT_XOREQ // ^=
TT_POINT, // ->
```

File: src/lexer/token.h [hunk 2]
```diff
@@ -17,6 +17,73 @@ typedef enum {
...
+	TT_BOREQ, // |=
+	TT_LANDEQ, // &&=
+	TT_LOREQ, // ||=
+	TT_GREATER, // >
+	TT_LESS, // <
+	TT_LESSEQ, // <=
+	TT_GREATEREQ, // >=
+	TT_LEFTSHIFT, // <<
+	TT_RIGHTSHIFT, // >>
+	TT_LNOT, // !
```
Comment (adamhutchings) [inline src/lexer/token.h]: Also this is minor, but tilde is a bitwise NOT, so if we want to have naming consistent we could call them LNOT and BNOT respectively.

File: src/lexer/token.h (hunk 1 above)
Comment (adamhutchings) [reply src/lexer/token.h]: Just realized -- also `TT_LEFTSHIFTEQUALS` and `TT_RIGHTSHIFTEQUALS` for `<<=` and `>>=`.

**Review by adamhutchings — CHANGES_REQUESTED** [review]
Thought of a couple more missing operators (thank you C, very cool)

File: src/lexer/token.h (hunk 1 above)
Comment (adamhutchings) [reply src/lexer/token.h]: Oh one more!!! (Hopefully this is the last one) there's also the dot operator `.`.

**Review by adamhutchings — COMMENTED** [review]
You mentioned you were going to add `ttype_from_string` support for all operators, so I will approve once that is finished

**Comment by JakeRoggenbuck** [comment]
> You mentioned you were going to add `ttype_from_string` support for all operators, so I will approve once that is finished

Okay, sounds good

**Comment by JakeRoggenbuck** [comment]
> > You mentioned you were going to add `ttype_from_string` support for all operators, so I will approve once that is finished
> 
> Okay, sounds good

STREQ has been added for all operators

**Review by adamhutchings — APPROVED** [review]

**Pull request merged**